### PR TITLE
jars: publish to s3

### PR DESCRIPTION
**Pull request by johnnyaug**

### CODE REVIEW

File: clients/spark/build.sbt [hunk 1]
```diff
@@ -89,9 +89,15 @@ lazy val commonSettings = Seq(
...
 lazy val publishSettings = Seq(
+  publishResolvers := Seq(
+    if (isSnapshot.value)
+      MavenRepository("lakefs-snapshots", s"$nexus/content/repositories/snapshots")
+    else MavenRepository("lakefs-releases", s"$nexus/service/local/staging/deploy/maven2"),
+    MavenRepository("lakefs-s3-repo", "s3://treeverse-clients-us-east/"),
+  ),
   publishTo := {
-    val nexus = "https://s01.oss.sonatype.org/"
     if (isSnapshot.value) Some("snapshots" at nexus + "content/repositories/snapshots")
```
Comment (arielshaqed) [inline clients/spark/build.sbt:101]: It is unfortunate that we need to say "snapshots" here and in the resolver settings.

File: clients/spark/project/plugins.sbt [hunk 2]
```diff
@@ -1,2 +1,4 @@
 addSbtPlugin("com.eed3si9n" % "sbt-assembly" % "0.15.0")
 addSbtPlugin("com.github.sbt" % "sbt-pgp" % "2.1.2")
+addSbtPlugin("laughedelic" % "sbt-publish-more" % "0.1.0")
```
Comment (arielshaqed) [inline clients/spark/project/plugins.sbt:3]: Do we still need this?

**Review by arielshaqed — COMMENTED** [review]
Cool, thanks!

File: clients/spark/build.sbt (hunk 1 above)
Comment (johnnyaug) [reply clients/spark/build.sbt:101]: Yes, although the `sbt-publish-more` should set the publishTo by default, which would remove the need to explicitly specify it - it doesn't work well with subprojects.

File: clients/spark/project/plugins.sbt (hunk 2 above)
Comment (johnnyaug) [reply clients/spark/project/plugins.sbt:3]: Yes - this is what allows us to publish to multiple repos (sonatype + s3)

**Review by arielshaqed — APPROVED** [review]
Cool, thanks!

**Pull request merged**